jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java

```diff
@@ -63,10 +63,12 @@ public void executePreparedStatement(PreparedStatement preparedStatement, boolea
     protected final Connection getConnection() {
         if (_connection == null) {
             _connection = getDataContext().getConnection();
-            try {
-                _connection.setAutoCommit(false);
-            } catch (SQLException e) {
-                throw JdbcUtils.wrapException(e, "disable auto-commit");
+            if (_dataContext.getQueryRewriter().isTransactional()) {
+                try {
```

Comment:
With this change we are now setting the autocommit to false only when the isTransactional condition is true, is this what we want?